docs/tutorials/craft-a-snap.rst

```diff
@@ -0,0 +1,549 @@
+      'block' font).
+
+      This snap is not endorsed by the pyfiglet project.
+
+As this is a personal snap, we prepended the project name with a user name. Replace
+``ukuzama`` with your own user name. You might encounter other snaps in the Snap Store
+with naming that follows this pattern -- it's the recommended format, to keep personal
+copies of snaps distinct from their originals.
+
+We didn't alter the ``base`` key, since we want our project to be built on top of the
+latest Ubuntu LTS release.
+
+Since we're packaging a project authored by someone else, we ought to respect their
+intentions and thinking in the description keys. So, we reused them. The ``summary`` is
+a short description with a hard limit of 79 characters, but nothing like that is present
+in the pyfiglet README. Instead, we sourced one from the upstream project at `figlet.org
+<https://www.figlet.org>`_. The fuller ``description`` key, which has no length limit,
+is taken from pyfiglet. We added a disclaimer about endorsement at the end.
+
+
+Define the target platforms
+---------------------------
+
+As we're building on an x64 system, and we're only running basic tests for this
+tutorial, we should constrain our snap to only build on the current CPU architecture,
+AMD64. At a later point, when you're able to test on other platforms, you could widen
+the coverage in this configuration.
+
+Add the ``platform`` key after the project information:
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+
+    platforms:
+      amd64:
+
+With this delcaration, Snapcraft will only build the snap on AMD64 machines, for AMD64
+machines. Take care to preserve the colon (:) in ``amd64:``.
+
+
+Define the main part
+--------------------
+
+A *part* is either a piece of software that we want to build from source or a clump of
+files. In either case, the purpose of a part is to bundle files from a source into a
+snap. Usually they're version-controlled components of the project itself, but you can
+aggregate them from a variety of locations.
+
+For the ``parts`` key, add an entry for our main part, the ``pyfiglet`` source code:
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+
+    parts:
+      pyfiglet:
+        plugin: python
+        source-type: git
+        source: https://github.com/pwaller/pyfiglet
+
+Parts have three important keys worth discussing.
+
+We set ``plugin`` to ``python`` because pyfiglet is a Python project and we want to
+build it from source.
+
+We set ``source-type`` to ``git`` because the project is stored as a Git repository.
+
+We set ``source`` to the remote location of the project. Some software projects take
+responsibility for their own snaps, and store their own ``snapcraft.yaml`` file in the
+source code. With pyfiglet, we're merely handling the packaging on the project's behalf,
+meaning our project file is downstream and dependent of it. By pointing to a remote URL,
```

Comment:
```suggestion
meaning our project file is downstream and independent of it. By pointing to a remote URL,
```

or?
```suggestion
meaning our project file is downstream and dependent on it. By pointing to a remote URL,
```

Reply:
I knew this one would come up. :) It's perfectly fine grammar, but I'll tweak it to be more globally friendly.

Reply:
50bb010d9c59071fe120f6b8924d569925f4299b